modules/telemetry/telemetry.php

```diff
@@ -0,0 +1,232 @@
+	 */
+	public static function record_update_custom_status( WP_Term $updated_status, array $update_args ): void {
+		$is_position_update = 1 === count( $update_args ) && isset( $update_args['position'] );
+		if ( $is_position_update ) {
+			// Ignore position changes, as they fire for every custom status when statuses are reordered
+			return;
+		}
+
+		self::$tracks->record_event( 'custom_status_changed', [
+			'term_id' => $updated_status->term_id,
+			'name'    => $updated_status->name,
+			'slug'    => $updated_status->slug,
+		] );
+	}
+
+	// Notification events
+
+	/**
+	 * Record an event when a notification is sent
+	 *
+	 * @param int $post_id The post ID of the post that was updated.
+	 * @param bool $is_email_scheduled True if an email was scheduled as part of the notification, false otherwise.
+	 * @param bool $is_webhook_scheduled True if a webhook was scheduled as part of the notification, false otherwise.
+	 */
+	public static function record_notification_sent( int $post_id, $is_email_scheduled, $is_webhook_scheduled ): void {
+		self::$tracks->record_event( 'notification_sent', [
+			'post_id' => $post_id,
+			'email'   => $is_email_scheduled,
+			'webhook' => $is_webhook_scheduled,
+		] );
+	}
+
+	// Settings events
+
+	/**
+	 * Record an event when the plugin is upgraded
+	 *
+	 * @param string $previous_version The previous version
+	 * @param string $new_version The new version
+	 */
+	public static function record_admin_update( string $previous_version, string $new_version ): void {
+		$custom_statuses      = CustomStatus::get_custom_statuses();
+		$supported_post_types = HelperUtilities::get_supported_post_types();
+
+		$custom_status_posts = 0;
+
+		foreach ( $supported_post_types as $post_type ) {
+			// Get all posts count for this post type
+			$posts_count = wp_count_posts( $post_type );
+
+			foreach ( $custom_statuses as $status ) {
+				if ( isset( $posts_count->{ $status->slug } ) ) {
+					$custom_status_posts += (int) $posts_count->{ $status->slug };
+				}
+			}
+		}
+
+		self::$tracks->record_event( 'plugin_update', [
+			'previous_version'    => $previous_version,
+			'new_version'         => $new_version,
+			'custom_status_posts' => $custom_status_posts,
+		] );
+	}
+
+	/**
+	 * Record presence of publish guard and webhook settings if toggled
+	 *
+	 * @param array $new_options The new options
+	 * @param array $old_options The old options
+	 */
+	public static function record_settings_update( array $new_options, array $old_options ): void {
+		if ( $new_options['publish_guard'] !== $old_options['publish_guard'] ) {
+			self::record_publish_guard_toggle( $new_options['publish_guard'] );
+		}
+
+		if ( $new_options['webhook_url'] !== $old_options['webhook_url'] ) {
+			self::record_send_to_webhook_toggle( $new_options['webhook_url'] );
+		}
+
+		if ( $new_options['email_address'] !== $old_options['email_address'] ) {
+			self::record_send_to_email_toggle( $new_options['email_address'] );
```

Comment:
Added this datapoint for tracking when a custom email address is added/deleted/changed, which works the same as the webhook URL toggle tracking.